sdk/eventgrid/eventgrid/src/cloudEventAdapter.ts

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { SendCloudEventInput } from "./models";
+
+interface MessageWithMetadata {
+  /**
+   * The message's binary data
+   */
+  body: Uint8Array;
+  /**
+   * The message's content type
+   */
+  contentType: string;
+}
+
+interface MessageAdapter<MessageT> {
+  /**
+   * defines how to create a message from a payload and a content type
+   */
+  produceMessage: (messageWithMetadata: MessageWithMetadata) => MessageT;
+  /**
+   * defines how to access the payload and the content type of a message
+   */
+  consumeMessage: (message: MessageT) => MessageWithMetadata;
+}
+
+type EncodedCloudEvent = SendCloudEventInput<Uint8Array>;
+
+function isUint8Array(obj: any): obj is Uint8Array {
+  return obj.constructor === Uint8Array;
```

Comment:
should we just use an `instanceof` check if we're going to be looking at the constructor property?

Reply:
Good point, I used `instanceof` and inlined it.